code/game/objects/items.dm

```diff
@@ -1092,4 +1092,69 @@ var/global/list/image/blood_overlays = list()
 	if(shrapnel_type)
 		return new shrapnel_type(src)
 	else
-		return 0
\ No newline at end of file
+		return 0
+
+/obj/item/duplicate(atom/destination)
+	var/obj/item/dupe = ..()
+
+	if(iscarbon(loc) && iscarbon(destination))
```

Comment:
Then I would recommend having each atom be able to provide a list of objects using a proc that should not be duplicated from its contents, and checking that in the parent proc. Then human and monkey code can handle inventory itself.